fontforge/ufo.c

```diff
@@ -2364,6 +2526,237 @@ static StemInfo *GlifParseHints(xmlDocPtr doc,xmlNodePtr dict,char *hinttype) {
 return( head );
 }
 
+static AnchorPoint *UFOLoadAnchor(SplineFont *sf, SplineChar *sc, xmlNodePtr xmlAnchor, AnchorPoint **lastap) {
+        xmlNodePtr points = xmlAnchor;
+        char *sname = (char *) xmlGetProp(points, (xmlChar *) "name");
+        if ( sname!=NULL) {
+
+            /* make an AP and if necessary an AC */
+            AnchorPoint *ap = chunkalloc(sizeof(AnchorPoint));
+            AnchorClass *ac;
+            char *namep = *sname=='_' ? sname + 1 : sname;
+            char *xs = (char *) xmlGetProp(points, (xmlChar *) "x");
+            char *ys = (char *) xmlGetProp(points, (xmlChar *) "y");
+            if (xs) { ap->me.x = strtod(xs,NULL); free(xs); }
+            if (ys) { ap->me.y = strtod(ys,NULL); free(ys); }
+
+            ac = SFFindOrAddAnchorClass(sf,namep,NULL);
+            if (*sname=='_')
+                ap->type = ac->type==act_curs ? at_centry : at_mark;
+            else
+                ap->type = ac->type==act_mkmk   ? at_basemark :
+                            ac->type==act_curs  ? at_cexit :
+                            ac->type==act_mklg  ? at_baselig :
+                                                  at_basechar;
+            ap->anchor = ac;
+	    if ( *lastap==NULL ) {
+			// If there are no existing anchors, we point the main spline reference to this one.
+			sc->anchor = ap;
+	    } else {
+			// If there are existing anchors, we attach to the last one.
+			(*lastap)->next = ap;
+	    }
+	    *lastap = ap;
+
+            free(sname);
+            return ap;
+        }
+        return NULL;
+}
+
+static real PointsSame(BasePoint p0, BasePoint p1) {
+	if (p0.x == p1.x && p0.y == p1.y)
+		return 1;
+	return 0;
+}
+
+static SplineSet *GuidelineToSpline(SplineFont *sf, GuidelineSet *gl) {
+	SplinePoint *sp1, *sp2;
+	SplineSet *ss;
+	real emsize = sf->ascent+sf->descent;
+	fprintf(stderr, "Em: %g.\n", emsize);
```

Comment:
Can we use the `TRACE` macros, or at least remove these?

Reply:
Done.